hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java

```diff
@@ -190,14 +201,44 @@ public static OzoneQuota parseQuota(String quotaInBytes,
           " value cannot be greater than Long.MAX_VALUE BYTES");
     }
 
-    if (nSize < 0) {
-      throw new IllegalArgumentException("Quota cannot be negative.");
+    if (nSize <= 0) {
+      throw new IllegalArgumentException("Invalid values for quota: " + nSize);
     }
 
-    return new OzoneQuota(quotaInNamespace,
-        new RawQuotaInBytes(currUnit, nSize));
+    return new OzoneQuota(new RawQuotaInBytes(currUnit, nSize));
   }
 
+  /**
+   * Parses a user provided string Namespace quota and returns the
+   * Quota Object.
+   *
+   * @param quotaInNamespace Volume quota in counts
+   *
+   * @return OzoneQuota object
+   */
+  public static OzoneQuota parseNameSpaceQuota(String quotaInNamespace) {
+    long nameSpaceQuota = Long.parseLong(quotaInNamespace);
+    if (nameSpaceQuota <= 0) {
```

Comment:
Question: how to set QUOTA.RESET? (basically how to disable quota by set the quota). In the past I guess we can set -2 but now it seems disallowed.

For example, is there a case that a bucket is enabled quota, later users decide to disable the quota by setting a quota value?

Reply:
If the user specifies quota when creating bucket and volume,the default is -1, indicating that quota is not enabled.

A quota value of -2 is a special case, because older clusters such as previous versions of Ozone1.0, volume Info and bucket Info do not have quotaInBytes and quotaInNamespace in them. After such a cluster is upgraded to Ozone1.1, these fields become to 0 when getting volume Info and bucket Info because they didn't exist before, so in HDDS-4562 proto is set to -2 by default, indicating that these volumes and buckets are old, and their quota is also not enabled. Currently, it is not recommended to enable quota for these volumes and buckets because their usedbyte and usedNamespace are incorrect. So, -2 is more used to distinguish between old volumes and buckets. Later if we can support recalculation of usedbyte and usedNamespace. This lets us know which volumes and buckets need to be recalculated.

To disable quota, use the clrquota method instead of setquota.